cheatsheets_draft/Cookie_Theft_Mitigation.md

```diff
@@ -0,0 +1,132 @@
+
+At the same time, even if the IP-Geo does not change, there is also the possibility that the attacker is attacking from within the same country. This means that this detection method has a **False positives** (it seems not to be an attack, but it is).
+
+### Cookie Theft Detection
+
+By storing session information on the server side when a session is established, it is possible to detect session hijacking when that information is significantly changed.
+
+The following are the core information that should be saved.
+
+- IP Address
+- User-Agent
+- Accept-Language
+- Date
+
+In addition, the following headers, which can be change depending on the Device and OS, are also effective as monitoring targets.
+
+- Accept
+- Accept-Encoding
+
+Also, recent browsers sends request headers called `Sec-Fetch-*` that provides information about the browsing contexts, so these values can also be used as a reference. It's not sent by the every browser, and it's not always sent even if browser supported. so it should be covered as Opt-In manner.
+
+- sec-ch-prefers-color-scheme
+- sec-ch-ua
+- sec-ch-ua-arch
+- sec-ch-ua-bitness
+- sec-ch-ua-form-factors
+- sec-ch-ua-full-version
+- sec-ch-ua-full-version-list
+- sec-ch-ua-mobile
+- sec-ch-ua-model
+- sec-ch-ua-platform
+- sec-ch-ua-platform-version
+- sec-ch-ua-wow64
+
+When a session is established on the server, this information is collected and saved in association with the session like below.
+
+```js
+const session = SessionStorage.create()
+session.save({
+  ip: req.clientIP,
+  user_agent: req.headers.userAgent,
+  date: req.headers.date,
+  accept_language: req.headers.acceptLanguage,
+  // ...
+})
+```
+
+If a large change is detected when comparing these informations each time a request is received, it is possible that the session has been hijacked.
+
+### Session Validation
+
+If there is a possibility that a session has been hijacked, the most reliable verification method is to re-authenticate. If you temporarily invalidate the user's session, ask them to authenticate again, and then give them a new session cookie, the attacker will no longer be able to do anything with the stolen cookie.
+
+However, as I mentioned earlier, monitoring sessions has the potential for false positives, so if you have to re-authenticate too often, it will be a bad experience for the user.
+
+An alternative would be to use a CAPTCHA or similar to make a decision. This is particularly useful when a stolen session cookie is being used by a bot or other malicious program.
+
+Basically, if there is a suspicion of session hijacking, it would be good to display a CAPTCHA for normal browsing, and to use re-authentication to provide reliable protection before accessing confidential information or performing processes with side effects.
```

Comment:
```suggestion
As a compromise, if there is a suspicion of session hijacking, it could be good practice to display a CAPTCHA for normal browsing, and to use re-authentication to provide reliable protection before accessing confidential information or performing actions with side effects.
```